### PR TITLE
Remove rebundant api named is_int_list

### LINKED ISSUE #114933: duplicate code in python_arg_parser.cpp
### 🐛 Describe the bug

‘static bool is_int_list(
    PyObject* obj,
    int broadcast_size,
    int64_t* failed_idx = nullptr) {
  if (PyTuple_Check(obj) || PyList_Check(obj)) {
    auto len = PySequence_Size(obj);
    if (len == 0) {
      return true;
    }

    auto item = py::reinterpret_steal<py::object>(PySequence_GetItem(obj, 0));
    bool int_first = false;
    if (THPUtils_checkIndex(item.ptr())) {
      // we still have to check that the rest of items are NOT symint nodes
      int_first = true;
    }

    // Make sure none of the later arguments are SymInt
    // NB: do NOT check that the later arguments are ints, as this is
    // BC-breaking for FX
    for (Py_ssize_t i = 1; i < len; i++) {
      if (torch::is_symint(
              py::reinterpret_steal<py::object>(PySequence_GetItem(obj, i)))) {
        if (failed_idx != nullptr) {
          *failed_idx = i;
        }
        return false;
      }
    }

    if (int_first) {
      return true;
    }

    // in dynamo, FakeTensor is qualified for INT_LIST
    if (is_dynamo_compiling && THPVariable_Check(item.ptr())) {
      auto& var = THPVariable_Unpack(item.ptr());
      if (var.numel() != 1 || !var.sizes().empty() ||
          !at::isIntegralType(
              var.dtype().toScalarType(), /*include_bool*/ true)) {
        if (failed_idx != nullptr) {
          *failed_idx = 0;
        }
        return false;
      }
      return true;
    }

    // NOTE: JIT tracer allows arbitrary scalar tensors to act as ints
    // in an intlist argument. Even float or complex scalar tensors.
    bool r =
        (jit::tracer::isTracing() && THPVariable_Check(item.ptr()) &&
         THPVariable_Unpack(item.ptr()).sizes().empty());
    if (!r && failed_idx != nullptr) {
      *failed_idx = 0;
    }
    return r;
  }
  // if a size is specified (e.g. IntArrayRef[2]) we also allow passing a single
  // int
  return broadcast_size > 0 && THPUtils_checkLong(obj);
}
‘
for above code, main part of it is duplicated with is_int_or_symint_list, except the `is_int_or_symint` check part.
it is annoying and hard to follow.

### Versions

branch main:
commit: [aa390ce](https://github.com/pytorch/pytorch/commit/aa390cec21108a17710be134190b4d7654222fed)

cc @ezyang @bhosmer @smessmer @ljk53 @bdhirsh

### PR DESCRIPTION
**Pull request by fffrog**
Fixes #114933

As the title stated.

### CODE REVIEW

**Comment by janeyx99** [comment]
@zou3519 assigning you as reviewer -- the PR looks fine to me, but I'm not sure if it's entirely okay to remove unused is_int_list (is it a private API)?

**Review by zou3519 — APPROVED** [review]
We don't maintain BC for our C++ API. As long as this passes CI (OSS + internal) I'm happy with yolo deleting it.
